### DISCUSSION TITLE
trying to compile model using omc, shows error

### DISCUSSION THREAD

**Discussion by ap766** [OpenModelica/OpenModelica#12406] [Q&A] [answered]
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/b7e1561e-24cd-4253-84e3-a21c1c09b918)
I am not able to fix this error

**Comment by perost** [comment]
Remove the `within Modules;` line at the top of the file, that's only for when the model is part of a package hierarchy and not when loading it as a standalone model.

**Reply by ap766** [reply]
So actually the Model Batch_Rectifier depends on another Compounds Model
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/1e9cf6e5-ad94-4d4f-bcd1-781d9b27b254)
after removing within modules this what I got 
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/caa12e1f-bac1-466f-9a5a-fd8abdae930b)

**Reply by ap766** [reply]
depends as in seen below
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/c7427ef1-1938-4d41-a489-7c94dc93d609)

**Reply by perost** [reply, marked as answer]
Then you need to load the whole library and not just that single model. If `Modules` is the top-level package then it would be:
```
omc package.mo -i=Modules.Batch_Rectifier -s
```
But guessing from the name I assume that `Modules` is maybe not the top-level package, so you might need to go further up in the hierarchy than that.

**Reply by ap766** [reply]
Thanks! It is the top level package in this case.

**Reply by ap766** [reply]
@perost  Also had a follow up question.So right now the c files are generated .I wanted to know how can I create .exe/.bat for the model now.
Thanks 

![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/1935f8b0-aa3e-47f3-9cb8-a22b03acc0be)
tion.

**Reply by ap766** [reply]
2)Another question - If it possible to convert the C code back into Modelica Code ? Any command for that?

**Reply by perost** [reply]
1. You need to use the generated makefile, `make -f Modules_Batch_Rectifier.makefile` (or whatever the makefile is called).
2. No, there is no way to decompile the C code into Modelica code.

**Reply by ap766** [reply]
Okay! Thank you

**Reply by casella** [reply]
Turning C code back into Modelica is like going against the 2nd principle of thermodynamics - during the conversion process from the equations to their solution some information about the original mathematical structure is lost and cannot be recovered.

**Reply by ap766** [reply]
Thats a nice analogy! 😄 Got it , thanks

**Reply by ap766** [reply]
@perost  i followed this `make -f Modules_Batch_Rectifier.makefile ` and seem to be getting the error below
![image](https://github.com/OpenModelica/OpenModelica/assets/79255079/c9869168-6c36-4253-9940-7274eb17fe74)
does that mean there's a issue with my .mo files?

**Reply by perost** [reply]
I'm guessing you're trying to compile some code that was generated by an older version of OM than you now have installed, in which case the header files of the new OM version might not be compatible with the generated code. In that case you'll have to recompile the model.
